pycbc/events/stat.py

```diff
@@ -845,13 +845,18 @@ def find_fits(self, trigs):
             The thresh fit value(s)
         """
         try:
-            tnum = trigs.template_num  # exists if accessed via coinc_findtrigs
+            tnum = trigs.template_num
+        except AttributeError:
+            tnum = trigs['template_id']
+
+        try:
             ifo = trigs.ifo
         except AttributeError:
-            tnum = trigs['template_id']  # exists for SingleDetTriggers
-            assert len(self.ifos) == 1
-            # Should be exactly one ifo provided
-            ifo = self.ifos[0]
+            ifo = trigs.get('ifo', None)
+            if ifo is None:
+                ifo = self.ifos[0]
+            assert ifo in self.ifos
```

Comment:
I cannot judge if this is correct, I do not understand what this bit is doing just by looking at it. Seems strange that the assertion is inside the except block though, is that intentional?

Reply:
I echo that this needs comments.

If trigs is dict-like you'll get to the except, then if `ifo` is not a key in the trigs object, then it could be met.

I am fairly sure that `self.ifos[0]` is _not_ the right thing to get here though, as that will always get the same ifo, even when you're using the triggers from a different detector.

Reply:
@GarethCabournDavies when do we expect trigs to be dict-like vs not? I've lost track

Reply:
Dict-like will be when it is called using:
 - a dictionary keyed on the dataset names like `pycbc_fit_sngls_by_template`
 - a SingleDetTriggers object, as in `pycbc_page_snglinfo` and `pycbc_sngl_minifollowup`
 - the h5py file directly as in `pycbc_fit_sngls_binned` (it shouldn't really do this, and should use SingleDetTriggers)

Not dictlike will be when using ReadByTemplate, which is done in `pycbc_coinc_findtrigs` and `pycbc_sngls_findtrigs`

Reply:
I was hoping there was some clean division like always dict-like when in PyCBC live and always not in offline, but I guess thats not the case. Might be nice to make that more consistent in the future, but not a task for this MR. From what I can see, I do think its the case that Live always uses the dict-like path?

Reply:
I think it is only used in live in the `single_stat = self.stat_calculator.single(trigsc)` line above, and `trigsc` is dictlike, so it seems this is safe